pkg/scheduler/framework/plugins/README.md

```diff
@@ -0,0 +1,141 @@
+type of the struct's fields may be pointers; leaving a parameter unspecified
+will set the pointer field to its zero value (nil), which can be used to let
+the framework know that it must fill in the default value. `BarParam` is of
+type `int32` and let's say we want a non-zero default value for it:
+
+```go
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type FooPluginArgs struct {
+	metav1.TypeMeta `json:",inline"`
+	BarParam *int32 `json:"barParam,omitempty"`
+}
+```
+
+For each `types_pluginargs.go` addition, remember to register the type in the
+corresponding `register.go`, which will allow the scheduler to recognize
+`KubeSchedulerConfiguration` values at parse-time.
+
+### Setting defaults
+
+When a `KubeSchedulerConfiguration` object is parsed (happens in
+`cmd/kube-scheduler/app/options/options.go`), the scheduler will convert from
+the latter type to the former type, filling in the unspecified fields with
+defaults. Speaking of defaults, define `SetDefaults_FooPluginArgs` in
+`pkg/scheduler/apis/config/v1beta1/defaults.go` as follows:
+
+```go
+// v1beta1 refers to k8s.io/kube-scheduler/config/v1beta1.
+func SetDefaults_FooPluginArgs(obj *v1beta1.FooPluginArgs) {
+	if obj.BarParam == nil {
+		obj.BarParam = pointer.Int32Ptr(42)
+	}
+}
+```
+
+### Validating configuration at runtime
+
+Next, we need to define validators to make sure the user's configuration and
+your default values are valid. To do this, add something like this in
+`pkg/scheduler/apis/config/validation/validation_pluginargs.go`:
+
+```go
+// From here on, FooPluginArgs refers to the type defined in pkg/scheduler
+// definition, not the kube-scheduler definition. We're dealing with
+// post-default values.
+func ValidateFooPluginArgs(args config.FooPluginArgs) error {
+	if args.BarParam < 0 && args.BarParam > 100 {
+		return fmt.Errorf("must be in the range [0, 100]")
+	}
+	return nil
+}
+```
+
+### Code generation
+
+We have defined everything necessary to run code generation now. Remember to
+commit all your changes (not sure why this is needed) and do a `make clean`
+first. Then:
+
+```sh
+$ cd $GOPATH/src/k8s.io/kubernetes
+$ git add -A && git commit
+$ make clean
+$ ./hack/update-codegen.sh
+$ make generated_files
+```
+
+This should automatically generate code to deep copy objects, convert between
+different struct types, convert pointer types to raw types, and set defaults.
+
+### Testing
+
+After code generation, go back and write tests for all of the changes you made
+in the previous section:
+
+- `pkg/scheduler/apis/config/v1beta1/defaults_test.go` to unit test the
+  defaults.
+- `pkg/scheduler/apis/config/validation/validation_pluginargs_test.go` to unit
+  test the validator.
+- `pkg/scheduler/apis/config/scheme/scheme_test.go` to test the whole pipeline
+  using a `KubeSchedulerConfiguration` definition.
+
+### Receiving the arguments in the plugin
+
+We can now finally receive `MyPluginArgs` in the plugin code. To do this,
```

Comment:
`FooPluginArgs`

Actually, it is very important that the name of the struct matches the name of the plugin (plus `Args`)